dace/transformation/dataflow/add_threadblock_map.py

```diff
@@ -0,0 +1,141 @@
+                                           allow_none=True,
+                                           desc="Number threads in the threadBlock Y Dim")
+    thread_block_size_z = SymbolicProperty(dtype=int,
+                                           default=None,
+                                           allow_none=True,
+                                           desc="Number threads in the threadBlock Z Dim")
+
+    @classmethod
+    def expressions(cls):
+        return [sdutil.node_path_graph(cls.map_entry)]
+
+    def preprocess_default_dims(self):
+        # If None is passed for the pass we will get the default configs
+        # 1. If arguments are passed:
+        #    1.1 Is the arguments passed
+        # 2. If no arguments are passed (at least one arg is None):
+        #   2.1. First check if the device map has gpu_block_size set
+        #   2.2. Otherwise check the global default
+        if self.thread_block_size_x is None or self.thread_block_size_y is None or self.thread_block_size_z is None:
+            if self.map_entry.gpu_block_size is not None:
+                # If gpu_block_size is set, use it
+                self.thread_block_size_x = self.map_entry.gpu_block_size[0]
+                self.thread_block_size_y = self.map_entry.gpu_block_size[1]
+                self.thread_block_size_z = self.map_entry.gpu_block_size[2]
+            else:
+                x, y, z = dace.config.Config.get('compiler', 'cuda', 'default_block_size').split(',')
+                try:
+                    self.thread_block_size_x = int(x)
+                    self.thread_block_size_y = int(y)
+                    self.thread_block_size_z = int(z)
+                except ValueError:
+                    raise ValueError("Invalid default block size format. Expected 'x,y,z' where x, y, z are integers.")
+
+            num_dims_in_map = len(self.map_entry.map.range)
+            # Collapse missing thread block dimensions into y if 2 dimensions in the map, to x if 1 dimension in the map
+            if num_dims_in_map < 3:
+                print_warning = False
+                old_block = (self.thread_block_size_x, self.thread_block_size_y, self.thread_block_size_z)
+                if num_dims_in_map == 2:
+                    self.thread_block_size_y *= self.thread_block_size_z
+                    if self.thread_block_size_z > 1:
+                        print_warning = True
+                    self.thread_block_size_z = 1
+                elif num_dims_in_map == 1:
+                    self.thread_block_size_x *= self.thread_block_size_y * self.thread_block_size_z
+                    if self.thread_block_size_y > 1 or self.thread_block_size_z > 1:
+                        print_warning = True
+                    self.thread_block_size_y = 1
+                    self.thread_block_size_z = 1
+                new_block = (self.thread_block_size_x, self.thread_block_size_y, self.thread_block_size_z)
```

Comment:
I am thinking that it would be simpler to write something similar to:
```python
old_block = (self.thread_block_size_x, self.thread_block_size_y, self.thread_block_size_z)
new_block = list(old_block)
for d in range(3, num_dims_in_map, -1):
    new_block[d-1] *= new_block[d]
    new_block[d] = 1
new_block = tuple(new_block)
if new_block != old_block:
    warnings.warn ...
```

It may not be immediately as readable as the current code though, so this just a suggestion.